src/main/kotlin/domain/usecase/GetRandomPotatoMealsUseCase.kt

```diff
@@ -0,0 +1,30 @@
+package org.example.domain.usecase
+
+import org.example.domain.FoodException
+import org.example.domain.repository.FoodRepository
+import org.example.model.Food
+
+class GetRandomPotatoMealsUseCase(private val repository: FoodRepository) {
+    companion object {
+        private const val POTATO = "potato"
+        private const val NUMBER_OF_MEALS = 10
+    }
+
+    fun getRandomPotatoMeals(): List<Food> {
+        val potatoMeals = getMealsContainingPotato()
+        if (potatoMeals.isEmpty()) throw FoodException.NoPotatoMealFound()
+        return potatoMeals.shuffled().take(NUMBER_OF_MEALS)
+    }
+
+    private fun getMealsContainingPotato(): List<Food> {
+        return repository.getFood()
+            .filter { food -> isMealContainsPotato(food) }
```

Comment:
I think we can use a function reference
```suggestion
                 .filter (::isMealContainsPotato)
```

Reply:
you right I'm resolved thanks a lot Mohamed